app/models/article.rb

```diff
@@ -567,6 +568,17 @@ def set_caches
     self.path = calculated_path.downcase
   end
 
+  def normalize_title
+    return unless title
+
+    self.title = title
+      # Filter out anything that isn't a word, space, punctuation mark, or
+      # recognized emoji
+      .gsub(/[^[:word:][:space:][:punct:]\u{1f000}-\u{1ffff}]+/, " ")
```

Comment:
If, and when, we need to reuse this regex pattern, it probably should have a name (either as a constant or an abstracted method).

This pattern of allowed characters sounds like it's the least certain part of the PR, and most likely to be required to change in the future, and may belong someplace outside of article (single source of truth, one reason to change).